lib/spack/spack/hooks/drop_redundant_rpaths.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from typing import Optional, Tuple
+
+import llnl.util.tty as tty
+from llnl.util.filesystem import BaseDirectoryVisitor, visit_directory_tree
+
+from spack.util.elf import ElfParsingError, parse_elf
+
+
+def should_keep(p: bytes):
```

Comment:
Can we use slightly more readable variable names:
```suggestion
def should_keep(path: bytes):
```
?